source/MaterialXRuntime/Private/PvtObject.h

```diff
@@ -76,9 +76,7 @@ class PvtObject : public RtRefCounted<PvtObject>
     {
         static_assert(std::is_base_of<PvtObject, T>::value,
             "Templated type must be an PvtObject or a subclass of PvtObject");
-#ifndef NDEBUG
-        // In debug mode we do safety checks on object validity
-        // and type cast compatibility.
+
```

Comment:
Turn this on all the time since it caught the logic error and is what was intended to give notice something is wrong.